### ISSUE TITLE
Improve overquota calculation

### ISSUE THREAD

**Issue by ethervoid** [CartoDB/cartodb#7023]
We have a clear performance problem with the `User.ovequota` method. With the current number of users it spents 1 hour and 40 minutes in process it all.

I've remove the mapviews calculation in order to reduce the time spent but we have to limit the number of users included inside this calculations. [Right now](https://github.com/CartoDB/cartodb/blob/master/app/models/user.rb#L452) we are getting all the activated and non-organizational users, we should add more filters like users with the soft_*_limit and/or with the twitter search configured.

It looks like the overquota method is only called from: an unused rake of mine that I should remove, and from Central to check the overquota. Only the users with soft limit and/or twitter search configured should be processed by overquota isn't it?

What do you think @juanignaciosl / @rafatower ?

**Comment by juanignaciosl** [comment]
You can't add the `soft_*` limit filters, because if it's disabled _now_ but it was enabled for a short period of time for a transient need we'd overlook them.

**Comment by juanignaciosl** [comment]
In fact I'm not sure about rejecting organization users.

What about doing this in parallel?

**Comment by ethervoid** [comment]
In the soft_* limit then the error comes from the checks for that. We should not let people has the soft_* and when the user is with overquota let the admin remove the soft_*. If this is happening now we should check it too.

We could process it in parallel but I think we should improve the single-thread version of the method because 167k users shouldn't be taking almost 2 hours to process.

**Comment by rafatower** [comment]
we need a quick solution now and then start thinking on a proper one for the future

(short/mid term future btw, we know for sure more services are to be added).

not quite the best repo for such ticket.

**Comment by xavijam** [comment]
There has been no activity on this issue for more several months. We are closing it. If you think this still needs to be addressed please open a new issue.
